content/concepts/did-ddo.md

```diff
@@ -1,173 +1,477 @@
+
+Example:
+
+```json
+{
+  "files": {
+    "files": [
+      {
+        "contentLength": "3975",
+        "contentType": "text/csv"
+      }
+    ],
+    "encryptedFiles": "0x044736da6dae39889ff570c34540f24e5e084f4e5bd81eff3691b729c2dd1465ae8292fc721e9d4b1f10f56ce12036c9d149a4dab454b0795bd3ef8b7722c6001e0becdad5caeb2005859642284ef6a546c7ed76f8b350480691f0f6c6dfdda6c1e4d50ee90e83ce3cb3ca0a1a5a2544e10daa6637893f4276bb8d7301eb35306ece50f61ca34dcab550b48181ec81673953d4eaa4b5f19a45c0e9db4cd9729696f16dd05e0edb460623c843a263291ebe757c1eb3435bb529cc19023e0f49db66ef781ca692655992ea2ca7351ac2882bf340c9d9cb523b0cbcd483731dc03f6251597856afa9a68a1e0da698cfc8e81824a69d92b108023666ee35de4a229ad7e1cfa9be9946db2d909735"
+  }
+}
+```
+
+## DDO Hash
+
+In order to ensure the integrity, a hash is computed for each DDO:
+
+```js
+const hash = sha256(JSON.stringify(DDO))
+```
+
+The hash is used when publishing/update metadata using setMetaData function in ERC721 contract and it is stored in the event generated by the ERC721 contract:
+
+```solidity
+event MetadataCreated(
+       address indexed createdBy,
+       uint8 state,
+       string decryptorUrl,
+       bytes flags,
+       bytes data,
+       bytes metaDataHash,
+       uint256 timestamp,
+       uint256 blockNumber
+   );
+
+event MetadataUpdated(
+       address indexed updatedBy,
+       uint8 state,
+       string decryptorUrl,
+       bytes flags,
+       bytes data,
+       bytes metaDataHash,
+       uint256 timestamp,
+       uint256 blockNumber
+   );
+```
+
+_Aquarius_ should always check the hash after data is decrypted via a _Provider_ API call, in order to ensure DDO integrity.
+
+## Aquarius Enhanced DDO Response
+
+The following fields are added by Aquarius in its DDO response for convenience reasons. These are never stored on chain, and not taken into consideration when [hashing the DDO](#ddo-hash).
+
+### Status
+
+The `status` object contains the following attributes:
+
+| Attribute             | Type      | Required | Description                                                               |
+| --------------------- | --------- | -------- | ------------------------------------------------------------------------- |
+| **`state`**           | `number`  | yes      | State of the asset reflecting the NFT contract value. See [State](#state) |
+| **`isListed`**        | `boolean` | no       | If this asset should be displayed                                         |
+| **`isOrderDisabled`** | `boolean` | no       | If this asset has ordering disabled                                       |
+
+```json
+{
+  "status": {
+    "state": 0,
+    "isListed": true,
+    "isOrderDisabled": false
+  }
+}
+```
+
+### Events
+
+The `events` section contains informations about the transactions that created or updated the DDO.
+
+Example:
+
+```json
+{
+  "events": [
```

Comment:
Note to implementers: I made this an array of events. If we manage to add the last event I'm sure we can manage to add all create/update events on here as only that makes it really useful for something like a metadata history.